vergen-git2/src/git2/mod.rs

```diff
@@ -150,17 +210,27 @@ pub struct Git2 {
     /// Optionally, add the `dirty` or `tags` flag to describe.
     /// See [`git describe`](https://git-scm.com/docs/git-describe#_options) for more details
     ///
-    #[builder(default = "false", setter(custom))]
-    describe: bool,
-    /// Instead of using only the annotated tags, use any tag found in refs/tags namespace.
-    #[builder(default = "false", private)]
-    describe_tags: bool,
-    /// If the working tree has local modification "-dirty" is appended to it.
-    #[builder(default = "false", private)]
-    describe_dirty: bool,
-    /// Only consider tags matching the given glob pattern, excluding the "refs/tags/" prefix.
-    #[builder(default = "None", private)]
-    describe_match_pattern: Option<&'static str>,
+    pub fn describe(mut self, tags: bool, dirty: bool, matches: Option<&'static str>) -> Self {
```

Comment:
Consider keeping the `describe` as a single field with a struct representation internally.

I had a branch where I tried to do these changes. Here is how I went about this: 

https://github.com/Veetaha/vergen/blob/b295ab148ee43933f620c2d411773507bcb13ee2/vergen-git2/src/git2/mod.rs#L165-L172

In general, I recommend you to take a look at https://github.com/rustyhorde/vergen/pull/395 to see some decisions I made there. For example, I added a field `#[builder(field)]` all: bool` which is used to control the default values of all other fields (the expression in `#[builder(default)]` has access to fields declared higher).